### PR TITLE
Hub: Modifies ui-new based on changes in upstream api response

### PR DESCRIPTION
**Pull request by pratap0007**
Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by PuneetPunamiya** [comment]
@pratap0007  Can we add the staging api url in the config file so that the application will show the resources when started locally as well

File: hub/ui-new/src/components/basic-detail/BasicDetail.tsx [hunk 1]
```diff
@@ -32,149 +32,180 @@ import './basicdetail.css';
...
     // eslint-disable-next-line
   }, [])
 
+
   const taskArr: any = [];
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  const [summary, setSummary] = useState(props.task.description.substring(0,
-    props.task.description.indexOf('\n')));
+  const [summary, setSummary] = useState(
```
Comment (sthaha) [inline hub/ui-new/src/components/basic-detail/BasicDetail.tsx:88]: these would read better if 
```
const {description} = props.task.latestVersion;
const summary = useState(description.substring(0, description.indexOf("\n")) || description)
```

File: hub/ui-new/src/components/basic-detail/BasicDetail.tsx [hunk 2]
```diff
@@ -32,149 +32,180 @@ import './basicdetail.css';
...
+    props.task.latestVersion.description.substring(0,
+      props.task.latestVersion.description.indexOf('\n')) ||
+    props.task.latestVersion.description,
+  );
 
   const [descrption, setDescription] =
     useState(
-      props.task.description.substring(props.task.description.indexOf('\n') +
-        1).trim());
+      props.task.latestVersion.description.indexOf('\n') !== -1 ?
```
Comment (sthaha) [inline hub/ui-new/src/components/basic-detail/BasicDetail.tsx:96]: indexOf("\n") is used at multiple places, perhaps it is better to store and reuse.

File: hub/ui-new/src/components/basic-detail/BasicDetail.tsx [hunk 3]
```diff
@@ -184,59 +215,89 @@ const BasicDetail: React.FC<BasicDetailProp> = (props: any) => {
...
-                  taskArr.map((tag: any) =>
-                    <Badge style={{paddingRight: '1em', marginBottom: '1em', marginRight: '1em'}}
-                      key={tag} className="badge">
-                      {tag}
-                    </Badge>)
-                }</GridItem>
+                <GridItem>
+                  {
+                    taskArr.map((tag: any) => {
+                      return (
```
Comment (sthaha) [inline hub/ui-new/src/components/basic-detail/BasicDetail.tsx:269]: `{ return( ` can be replaced by `(`
e.g.
 
```
const foo = () => { 
  return (
    <Foo>
       <bar />
    </Foo>
  )
}
```
can be better written as 

```
const foo = () => (
   <Foo>
     <bar/>
   </Foo>
)
```

File: hub/ui-new/src/components/filter/Sort.tsx [hunk 4]
```diff
@@ -4,34 +4,50 @@ import {connect} from 'react-redux';
...
+}
+
+// export interface TaskPropData{
+//   task: TaskPropObject
+// }
 const Sort: React.FC = (props: any) => {
   const [sort, setSort] = useState('Name');
   let tempArr: any = [];
   if (props.TaskData) {
     tempArr = props.TaskData.map((task: any) => {
```
Comment (sthaha) [inline hub/ui-new/src/components/filter/Sort.tsx:42]: why is the tempArr needed here?

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**